src/js/components/TaskDetailComponent.jsx

```diff
@@ -41,10 +42,33 @@ var TaskDetailComponent = React.createClass({
   getTaskEndpoints: function () {
     var props = this.props;
     var task = props.task;
-    if (task.ports == null || task.ports.length === 0) {
+    var app = AppsStore.getCurrentApp(props.appId);
+
+    if ((task.ports == null || task.ports.length === 0) &&
+        (task.ipAddresses == null || task.ipAddresses.length === 0)) {
       return (<dd>None</dd>);
     }
 
+    if (app.ipAddress != null &&
```

Comment:
After reviewing this PR line by line, I would like to have this with ```objectPath.get(app, "ipAddress.discovery.ports")```. You can see much better then what's going on. I've tried it.
Could yoiu refactor it? 
Sorry for the see-saw!